pkg/connector/users.go

```diff
@@ -87,10 +90,66 @@ func (o *UserBuilder) Entitlements(_ context.Context, resource *v2.Resource, _ *
 	return nil, "", nil, nil
 }
 
-// Grants always returns an empty slice for users since they don't have any entitlements.
-func (o *UserBuilder) Grants(ctx context.Context, resource *v2.Resource, pToken *pagination.Token) ([]*v2.Grant, string, annotations.Annotations, error) {
-	return nil, "", nil, nil
+// Role grants are implemented here for performance reasons.
+func (o *UserBuilder) Grants(ctx context.Context, res *v2.Resource, _ *pagination.Token) ([]*v2.Grant, string, annotations.Annotations, error) {
+	l := ctxzap.Extract(ctx)
+	userID := res.Id.Resource
+
+	user, err := o.client.GetUser(ctx, userID)
+	if err != nil {
+		l.Error("failed to fetch user for grant resolution", zap.String("user_id", userID), zap.Error(err))
+		return nil, "", nil, err
+	}
+
+	var grants []*v2.Grant
+
+	// BaseRole
+	if user.BaseRole.ID != "" {
+		baseRoleResource := &v2.Resource{
+			Id: &v2.ResourceId{
+				ResourceType: baseRoleResourceType.Id,
+				Resource:     user.BaseRole.ID,
+			},
+		}
+
+		grant := grant.NewGrant(
+			baseRoleResource,
+			roleGrantPermission,
+			res,
+			grant.WithAnnotation(&v2.V1Identifier{
+				Id: fmt.Sprintf("base-role-grant:%s:%s", user.BaseRole.ID, userID),
+			}),
+		)
+		grants = append(grants, grant)
+	}
+
+	// CustomRoles
+	for _, cr := range user.CustomRoles {
+		if cr.ID == "" {
+			continue
+		}
+
+		customRoleResource := &v2.Resource{
+			Id: &v2.ResourceId{
+				ResourceType: customRoleResourceType.Id,
+				Resource:     cr.ID,
+			},
+		}
+
+		grant := grant.NewGrant(
+			customRoleResource,
+			roleGrantPermission,
+			res,
+			grant.WithAnnotation(&v2.V1Identifier{
+				Id: fmt.Sprintf("custom-role-grant:%s:%s", cr.ID, userID),
```

Comment:
What's the V1 identifier for here? I thought V1 identifiers were meant to make it easier to migrate V1 connectors (non-baton). We don't have a v1 connector for Incident.io, so there's not much of a point in using this annotation.